vnext/Microsoft.ReactNative/Views/TextInputViewManager.cpp

```diff
@@ -802,8 +827,43 @@ ShadowNode *TextInputViewManager::createShadow() const {
   return new TextInputShadowNode();
 }
 
-XamlView TextInputViewManager::CreateViewCore(int64_t /*tag*/, const winrt::Microsoft::ReactNative::JSValueObject &) {
+XamlView TextInputViewManager::CreateViewCore(int64_t tag, const winrt::Microsoft::ReactNative::JSValueObject &) {
   xaml::Controls::TextBox textBox;
+  // This works around a XAML Islands bug where the XamlRoot of the first
+  // window the flyout is shown on takes ownership of the flyout and attempts
+  // to show the flyout on other windows cause the first window to get focus.
+  // https://github.com/microsoft/microsoft-ui-xaml/issues/5341
+  if (IsXamlIsland() &&
+      winrt::Windows::Foundation::Metadata::ApiInformation::IsApiContractPresent(
+          L"Windows.Foundation.UniversalApiContract", 7)) {
```

Comment:
I think we will want to tailor this check a bit and avoid doing this work in some cases, let's follow up offline :)

Reply:
We could also add a property to check if the app is a single-threaded multi-window app, as this issue only occurs in that case.